api/v1alpha3/azurecluster_validation.go

```diff
@@ -29,7 +29,7 @@ import (
 const (
 	// can't use: \/"'[]:|<>+=;,.?*@&, Can't start with underscore. Can't end with period or hyphen.
 	// not using . in the name to avoid issues when the name is part of DNS name
-	clusterNameRegex = `^[a-z][a-z0-9-]{0,44}[a-z0-9]$`
+	clusterNameRegex = `^[a-z0-9][a-z0-9-]{0,44}[a-z0-9]$`
```

Comment:
let's add a unit test case that validates this change

Reply:
done